source.lua

```diff
@@ -1,38 +1,68 @@
 --@ module = true
 local repeatUtil = require('repeat-util')
+local json = require('json')
+local persist = require('persist-table')
 
-liquidSources = liquidSources or {}
+local GLOBAL_KEY = 'source' -- used for state change hooks and persistence
 
 local sourceId = 'liquidSources'
```

Comment:
you can get rid of `sourceId` and use `GLOBAL_KEY` for this as well. no need to have two separate and different IDs